api/middleware.go

```diff
@@ -18,12 +19,37 @@ const (
 	jwsSignatureHeaderName = "x-nf-sign"
 )
 
+type FunctionHooks map[string][]string
+
 type NetlifyMicroserviceClaims struct {
 	jwt.StandardClaims
-	SiteURL       string            `json:"site_url"`
-	InstanceID    string            `json:"id"`
-	NetlifyID     string            `json:"netlify_id"`
-	FunctionHooks map[string]string `json:"function_hooks"`
+	SiteURL       string        `json:"site_url"`
+	InstanceID    string        `json:"id"`
+	NetlifyID     string        `json:"netlify_id"`
+	FunctionHooks FunctionHooks `json:"function_hooks"`
+}
+
+func (f FunctionHooks) UnmarshalJSON(b []byte) error {
```

Comment:
could you add a test for this?

Reply:
Thanks for suggesting it, there were some problems with the original.